pkg/controller/kfservice/resources/knative/service.go

```diff
@@ -87,15 +98,22 @@ func (c *ServiceBuilder) CreateEndpointService(kfsvc *v1alpha2.KFService, endpoi
 		//TODO create transformer
 		return nil, nil
 	case constants.Explainer:
-		explainerSpec := &kfsvc.Spec.Default.Explainer
+		explainerSpec := kfsvc.Spec.Default.Explainer
+		predictorHost := constants.DefaultPredictorServiceName(kfsvc.Name) + "." + kfsvc.ObjectMeta.Namespace
+		if isCanary {
+			predictorHost = constants.CanaryPredictorServiceName(kfsvc.Name) + "." + kfsvc.ObjectMeta.Namespace
```

Comment:
I think kfsvc.Namespace is ok as kfsvc.Name instead of kfsvc.ObjectMeta.Namespace